src/main.c

```diff
@@ -7,56 +7,40 @@
 #include "mmu.h"
 #include "cli.h"
 #include "logging.h"
-#include "errcodes.h"
 
 int main(const int argc, char **const argv) {
-    CLIArguments* args = cli_config_default();
-    
-    if (cli_config_handle(args, argc, argv)) {
-        cli_config_destroy(args);
-        exit(ERR_FAILURE);
-    }
+	CLIArguments cli_args;
+	cli_parse(&cli_args, argc, argv);
 
-    ErrorCode ret = rom_load("./roms/hello-world.gb");
-    if (ret) {
-        LOG_FATAL("Failed to load ROM");
-        exit(ret);
-    }
-    printf("Successfully initialized ROM\n");
+	log_set_lvl(cli_args.logging_level);
```

Comment:
I like this better than what I have implemented. Keeps the modules nice and separated.

Reply:
Thank you, very kind of you to say that!